### PR TITLE
Bdog 622

### PR DESCRIPTION
**Pull request by colin-lamed**

### CODE REVIEW

File: app/uk/gov/hmrc/teamsandrepositories/services/GithubV3RepositoryDataSource.scala [hunk 1]
```diff
@@ -102,31 +104,31 @@ class GithubV3RepositoryDataSource(
...
   def getRepositoryDetailsFromGithub(repository: GhRepository): Future[GitRepository] =
     for {
-      manifest                <- withCounter(s"github.open.fileContent") {
-                                   gh.getFileContent("repository.yaml", repository.name, HMRC_ORG)
-                                 }
-      maybeManifestDetails    =  getMaybeManifestDetails(repository.name, manifest)
-      repositoryType          <- identifyRepository(repository, maybeManifestDetails.flatMap(_.repositoryType))
-      maybeDigitalServiceName =  maybeManifestDetails.flatMap(_.digitalServiceName)
-      owningTeams             =  maybeManifestDetails.map(_.owningTeams).getOrElse(Nil)
+      optManifest     <- githubConnector.getFileContent(repository.name, "repository.yaml")
```
Comment (bennetimo) [inline app/uk/gov/hmrc/teamsandrepositories/services/GithubV3RepositoryDataSource.scala]: is it intentional to leave out the counter?

File: app/uk/gov/hmrc/teamsandrepositories/connectors/GithubConnector.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, NotFoundException}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.teamsandrepositories.config.GithubConfig
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GithubConnector @Inject()(githubConfig: GithubConfig, http: HttpClient)(implicit ec: ExecutionContext) {
+
+  def getFileContent(repoName: String, path: String): Future[Option[String]] = {
+    implicit val hc = HeaderCarrier().withExtraHeaders("Authorization" -> s"token ${githubConfig.githubApiOpenConfig.key}")
```
Comment (christopherjturner) [inline app/uk/gov/hmrc/teamsandrepositories/connectors/GithubConnector.scala]: I might be mis-remembering this but does http verbs do something weird when setting the Authorization header? I seem to remember there being some issues with the artifactory connector back in the day?

File: app/uk/gov/hmrc/teamsandrepositories/services/GithubV3RepositoryDataSource.scala (hunk 1 above)
Comment (colin-lamed) [reply app/uk/gov/hmrc/teamsandrepositories/services/GithubV3RepositoryDataSource.scala]: Yes, the metrics were for the `github.open` api usage.

**Comment by colin-lamed** [comment]
superseded by BDOG-572

**Pull request closed without merging**